command/agent/command.go

```diff
@@ -321,6 +321,13 @@ func (c *Command) readConfig() *Config {
 		c.Ui.Error("WARNING: Bootstrap mode enabled! Potentially unsafe operation.")
 	}
 
+	// Set up the TLS configuration properly if we have one.
+	// XXX chelseakomlo: set up a TLSConfig New method which would wrap
+	// constructor-type actions like this.
+	if config.TLSConfig != nil && !config.TLSConfig.IsEmpty() {
+		config.TLSConfig.SetChecksum()
```

Comment:
Capture the err and if it isn't nil output the error with a warning like: https://github.com/hashicorp/nomad/pull/4046/files#diff-9d1ee8f5adf90f07803d3f8bbc339694L321